core/src/main/scala/org/apache/spark/sql/streaming/SnappySinkCallback.scala

```diff
@@ -163,6 +162,7 @@ class DefaultSnappySinkCallback extends SnappySinkCallback {
         s", eventTypeColumnAvailable:$eventTypeColumnAvailable,possible duplicate: $posDup")
 
     if (keyColumns.nonEmpty) {
+      df.cache().count()
```

Comment:
There is a way to avoid creating unnecessary tasks in the SQL tab. See SnappyCacheTableCommand:

```scala
        val memoryPlan = df.queryExecution.executedPlan.collectFirst {
          case plan: InMemoryTableScanExec => plan.relation
        }.get
        memoryPlan.cachedColumnBuffers.count()
```

On the other hand if we do want to show the incoming DF plan that got cached, then the df.cache().count() won't show it rather shows only the count aggregate plan on InMemoryRelation after it has been created. So I think it will be better to refactor SnappyCacheTableCommand from the portion after DF is created in first few lines, and use here which will show the incoming DF plan that matters and also take care of using off-heap for the cache if configured.

Reply:
Tracked by https://jira.snappydata.io/browse/SNAP-2761.